core/src/main/java/me/leoko/advancedban/utils/tabcompletion/PunishmentTabCompleter.java

```diff
@@ -20,6 +20,10 @@ public List<String> onTabComplete(Object user, String[] args) {
         final MethodInterface methodInterface = Universal.get().getMethods();
 
         List<String> suggestions = new ArrayList<>();
+        
+        if (!methodInterface.getBoolean(methodInterface.getConfig(), "Use Tab Completion", true)) {
```

Comment:
PunishmentTabCompleter is only one implementation of TabCompleter. To block all tab completion, you'd need to move this check further up the call chain.